src/ATen/native/xpu/BinaryOps.cpp

```diff
@@ -454,4 +455,60 @@ Tensor XPUNativeFunctions::sigmoid_backward(
   return iter.output();
 }
 
+Tensor& XPUNativeFunctions::copysign_out(
+    const Tensor& self,
+    const Tensor& other,
+    Tensor& out) {
+  Tensor other_maybe_scalar = other;
+  if (other.device().type() == at::kCPU && other.numel() == 1) {
+    other_maybe_scalar = other.to("xpu");
+  }
+
+  TensorIterator iter;
+  iter.build_borrowing_binary_float_op(out, self, other_maybe_scalar);
+  native::xpu::copysign_kernel(iter);
+  return out;
+}
+
+Tensor& XPUNativeFunctions::copysign_(Tensor& self, const Tensor& other) {
+  return XPUNativeFunctions::copysign_out(self, other, self);
+}
+
+Tensor XPUNativeFunctions::copysign(const Tensor& self, const Tensor& other) {
+  Tensor out;
+  Tensor other_maybe_scalar = other;
+  if (other.device().type() == at::kCPU && other.numel() == 1) {
+    other_maybe_scalar = other.to("xpu");
+  }
+
+  TensorIterator iter;
+  iter.build_borrowing_binary_float_op(out, self, other_maybe_scalar);
+  native::xpu::copysign_kernel(iter);
+  return iter.output();
+}
+
+Tensor& XPUNativeFunctions::copysign_out(
+    const Tensor& self,
+    const Scalar& other,
+    Tensor& out) {
+  auto wrapper = native::wrapped_scalar_tensor(other, self.device());
+  TensorIterator iter;
+  iter.build_borrowing_binary_float_op(out, self, wrapper);
+  native::xpu::copysign_kernel(iter);
+  return out;
+}
+
+Tensor XPUNativeFunctions::copysign(const Tensor& self, const Scalar& other) {
+  Tensor out;
+  auto wrapper = native::wrapped_scalar_tensor(other, self.device());
+  TensorIterator iter;
+  iter.build_borrowing_binary_float_op(out, self, wrapper);
+  native::xpu::copysign_kernel(iter);
+  return iter.output();
+}
+
+Tensor& XPUNativeFunctions::copysign_(Tensor& self, const Scalar& other) {
+  return XPUNativeFunctions::copysign_out(self, other, self);
+}
+
```

Comment:
Scalar version operators should not be registered.
<img width="658" alt="image" src="https://github.com/intel/torch-xpu-ops/assets/28250760/5a690c5e-e946-4743-9874-e2bdeddbd8e0">